include/ddc/kernels/splines/matrix_sparse.hpp

```diff
@@ -122,17 +122,20 @@ class Matrix_Sparse : public Matrix
         std::shared_ptr const gko_exec = create_gko_exec<ExecSpace>();
         m_matrix_dense = gko::matrix::Dense<
                 double>::create(gko_exec->get_master(), gko::dim<2>(mat_size, mat_size));
-        m_matrix_dense->fill(0);
         m_matrix_sparse = matrix_sparse_type::create(gko_exec, gko::dim<2>(mat_size, mat_size));
+
+        m_matrix_dense->fill(0);
+        m_matrix_sparse->clear();
     }
 
-    virtual double get_element([[maybe_unused]] int i, [[maybe_unused]] int j) const override
+    double get_element([[maybe_unused]] int i, [[maybe_unused]] int j) const override
     {
         throw std::runtime_error("MatrixSparse::get_element() is not implemented because no API is "
                                  "provided by Ginkgo");
+        return 0;
```

Comment:
```suggestion
```

harmless but not required

Reply:
dont you think it will trigger a warning at compile time ?

Reply:
Should not, to my knowledge compilers handle fine the fact that there is a missing return after a `throw`